### PR TITLE
Update OptionsArray phpdoc & add tests for `SET GLOBAL`

### PR DESCRIPTION
**Pull request by ansavin**
Fixed type comment in order to pass linter check while fixing issue#17951 in phpmyadmin

Linter error: `Cannot fetch property on non-object $statement->options of type array<array-key, PhpMyAdmin\SqlParser\Components\OptionsArray>`
(example in https://github.com/phpmyadmin/phpmyadmin/actions/runs/4780247897/jobs/8498639908?pr=18349)

See https://github.com/phpmyadmin/phpmyadmin/pull/18349 for more info

### CODE REVIEW

**Comment by williamdes** [comment]
Thanks @ansavin
Can you rebase your branch onto 5.8.x?
Please do not close or open a new PR

File: src/Statements/SetStatement.php [hunk 1]
```diff
@@ -82,7 +82,7 @@ class SetStatement extends Statement
     /**
      * Options used in current statement.
      *
-     * @var OptionsArray[]|null
+     * @var OptionsArray|null
```
Comment (williamdes) [inline src/Statements/SetStatement.php:85]: Needs to be checked if that's a mistake or an actual existing behavior

**Comment by ansavin** [comment]
> Thanks @ansavin Can you rebase your branch onto 5.8.x? Please do not close or open a new PR

No problem, I done it just recently

File: src/Statements/SetStatement.php (hunk 1 above)
Comment (williamdes) [inline src/Statements/SetStatement.php:85]: For the record it was added in https://github.com/phpmyadmin/sql-parser/pull/75

I see no reason for it to be an array

**Pull request merged**